apps/site/util/authorUtils.ts

```diff
@@ -0,0 +1,73 @@
+import authors from '@/authors.json';
+
+import { getGitHubAvatarUrl } from './gitHubUtils';
+import { getAcronymFromString } from './stringUtils';
+
+export const mapAuthorToCardAuthors = (author: string) => {
+  // Clears text in parentheses
+  const cleanedAuthor = author.replace(/\s*\(.*?\)\s*/g, '').trim();
+
+  // Defines the separators such as (",", "and", ";", "&", "prepared by", "by")
+  const separators = /,|\band\b|;|&| prepared by | by /i;
+
+  return cleanedAuthor
+    .split(separators)
+    .map(name => name.trim())
+    .filter(Boolean);
+};
+
+type Author = {
+  id: string;
+  name: string;
+  website?: string;
+};
+
+export const getAuthorWithId = (usernames: Array<string>, hasUrl: boolean) =>
+  usernames.map(username => {
+    const author = Object.values(authors).find(
+      ({ id }: { id: string }) => id.toLowerCase() === username.toLowerCase()
```

Comment:
This should already be typed from the json import by moving the json import to the file where we import all json files